### PR TITLE
Instructions aggregator json

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Provide a way to get a JSON summary of relevant info from a stage (to be consumed by JoshC)

### CODE REVIEW

File: dashboard/app/controllers/script_levels_controller.rb [hunk 1]
```diff
@@ -127,6 +127,58 @@ def toggle_hidden
...
+
+    if params[:stage_position]
+      stage = script.stages.select{|s| !s.lockable? && s.relative_position == params[:stage_position].to_i }.first
+    else
+      stage = script.stages.select{|s| s.lockable? && s.relative_position == params[:lockable_stage_position].to_i }.first
+    end
+
+    render json: {
+      stageName: stage.name,
+      lockable: stage.lockable? ? true : false,
```
Comment (joshlory) [inline dashboard/app/controllers/script_levels_controller.rb]: Isn't `stage.lockable?` already a boolean?

File: dashboard/app/controllers/script_levels_controller.rb [hunk 2]
```diff
@@ -127,6 +127,58 @@ def toggle_hidden
...
+    script = Script.get_from_cache(params[:script_id])
+
+    if params[:stage_position]
+      stage = script.stages.select{|s| !s.lockable? && s.relative_position == params[:stage_position].to_i }.first
+    else
+      stage = script.stages.select{|s| s.lockable? && s.relative_position == params[:lockable_stage_position].to_i }.first
+    end
+
+    render json: {
+      stageName: stage.name,
```
Comment (joshlory) [inline dashboard/app/controllers/script_levels_controller.rb]: `stage.localized_name`

File: dashboard/app/controllers/script_levels_controller.rb [hunk 3]
```diff
@@ -127,6 +127,58 @@ def toggle_hidden
...
+    require_levelbuilder_mode
+    authorize! :read, ScriptLevel
+
+    script = Script.get_from_cache(params[:script_id])
+
+    render 'levels/instructions', locals: { stages: script.stages }
+  end
+
+  # Provides a JSON summary of a particular stage
+  def summary
```
Comment (joshlory) [inline dashboard/app/controllers/script_levels_controller.rb]: How is this different from `Script#summarize`?

File: dashboard/app/controllers/script_levels_controller.rb (hunk 1 above)
Comment (Bjvanminnen) [reply dashboard/app/controllers/script_levels_controller.rb]: yeah, good point

File: dashboard/app/controllers/script_levels_controller.rb (hunk 3 above)
Comment (Bjvanminnen) [reply dashboard/app/controllers/script_levels_controller.rb]: It exposes a different set of information. I was struggling with this a little myself. Perhaps one solution would be to name this something like lesson_plan_summary and add a comment making it clear that this is what it's used for? Still leaves us in the unfortunate situation of having two different summaries though.

Incidentally, I also rather dislike `Script#summarize` and some of the stuff it passes down (for example icon-names).

File: dashboard/app/models/script_level.rb [hunk 4]
```diff
@@ -166,6 +166,10 @@ def stage_total
     stage.script_levels.to_a.size
   end
 
+  def path
+    build_script_level_path(self)
```
Comment (joshlory) [inline dashboard/app/models/script_level.rb:170]: Weird, we include `LevelsHelper` in this model 😕.

File: dashboard/app/models/stage.rb [hunk 5]
```diff
@@ -127,6 +127,37 @@ def summarize
     stage_summary.freeze
   end
 
+  # Provides a JSON summary of a particular stage, that is consumed by tools used to
+  # build lesson plans
+  def summary_for_lesson_plans
```
Comment (joshlory) [inline dashboard/app/models/stage.rb:132]: It still seems not great to have this method in addition to `Stage#summarize` but I'm not sure how I'd de-duplicate.

**Comment by joshlory** [comment]
LGTM

**Pull request merged**